sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -2422,6 +2422,12 @@ class ReadFromBigQuery(PTransform):
       to run queries with INTERACTIVE priority. This option is ignored when
       reading from a table rather than a query. To learn more about query
       priority, see: https://cloud.google.com/bigquery/docs/running-queries
```

Comment:
I looked over the other parameters here, I think there are a couple of cases that we will need to reject when they are used in conjunction with `output_type=BEAM_ROW`:
- If `table` is a callable or a ValueProvider
- If `query` is used instead of project/dataset/table

In both of these cases we can't (easily) resolve the schema at pipeline construction time. I think as written now we will just fail anyway when either of these happen, but it won't be a helpful error. We should detect these cases and raise a nice error (and test that we do) and document the limitation.

(In theory we should be able to determine the schema for the `query` case too, but that's a little trickier. I think it's a good item to leave for future work)